AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -4019,17 +4016,26 @@ function Debug-AzStorageAccountEntraKerbAuth {
                     -ResourceGroupName $ResourceGroupName `
                     -StorageAccountName $StorageAccountName `
                     -ErrorAction Stop
- 
-                $DefaultSharePermission = $StorageAccountObject.AzureFilesIdentityBasedAuth.DefaultSharePermission
                 
-                # If DefaultSharePermission is null or 'None'
-                if((!$DefaultSharePermission) -or ($DefaultSharePermission -eq 'None')){
-                    Debug-RBACCheck -StorageAccountName $StorageAccountName -checkResult $checks["CheckDefaultShareRBACPermission"]
-                    #$DefaultSharePermission = "Not Configured. Please visit https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-ad-ds-assign-permissions?tabs=azure-portal for more information if needed."
+                if ($null -eq $StorageAccountObject.AzureFilesIdentityBasedAuth)
+                { 
+                    $checks["CheckDefaultShareRBACPermission"].Result = "Failed"
+                    $checks["CheckDefaultShareRBACPermission"].Issue = "AzureFilesIdentityBasedAuth IS NULL"
+                    Write-Error "CheckDefaultShareRBACPermission - FAILED"
+                }
+                else 
+                {
+                    $DefaultSharePermission = $StorageAccountObject.AzureFilesIdentityBasedAuth.DefaultSharePermission
+                    
+                    if((!$DefaultSharePermission) -or ($DefaultSharePermission -eq 'None'))
+                    {
+                        Debug-RBACCheck -StorageAccountName $StorageAccountName -UserPrincipalName $UserPrincipalName -checkResult $checks["CheckDefaultShareRBACPermission"]
+                        #$DefaultSharePermission = "Not Configured. Please visit https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-ad-ds-assign-permissions?tabs=azure-portal for more information if needed."
+                    }
+                    Write-Verbose "CheckDefaultShareRBACPermission: $CheckDefaultShareRBACPermission"
+                    Write-Verbose "CheckDefaultShareRBACPermission - SUCCESS"
+                    $checks["CheckDefaultShareRBACPermission"].Result = "Passed"

```

Comment:
If `Debug-RBACCheck` is responsible for writing failures in the `CheckResult` object, then I suggest we also move the responsibility of writing successes into that function.